crates/components-runtime/src/data_collection/mod.rs

```
@@ -78,6 +85,96 @@ pub async fn send_json_events(
+        match crate::data_collection::versions::v1_0_1::execute::get_auth_request(
+            context, component, &mut store,
+        )
+        .await
+        {
+            Ok((headers, method, url, body, auth_metadata)) => {
+                (headers, method, url, body, auth_metadata)
+            }
+            Err(err) => {
+                error!("Failed to get auth request. Error: {}", err);
+                return Err(err);
+            }
+        };
+
+    let client = reqwest::Client::builder()
+        .timeout(Duration::from_secs(5))
+        .build()?;
+
+    let component_id = component.id.clone();
+    let future = tokio::spawn(async move {
+        let res = match method.as_str() {
+            "GET" => client.get(url.clone()).headers(headers).send().await,
+            "PUT" => {
+                client
+                    .put(url.clone())
+                    .headers(headers)
+                    .body(body.clone())
+                    .send()
+                    .await
+            }
+            "POST" => {
+                client
+                    .post(url.clone())
+                    .headers(headers)
+                    .body(body.clone())
+                    .send()
+                    .await
+            }
+            _ => {
+                return AuthResponse {
+                    component_id,
+                    token: String::new(),
+                    token_duration: auth_metadata.token_duration,
+                }
+            }
+        };
+
+        let response_text = match res {
+            Ok(r) => r.text().await.unwrap_or_default(),
+            Err(_) => String::new(),
+        };
+
+        let token = serde_json::from_str::<serde_json::Value>(&response_text)
+            .ok()
+            .and_then(|json| {
+                json.get(&auth_metadata.token_property_name)
+                    .and_then(|v| v.as_str().map(|s| s.to_string()))
+            })
+            .unwrap_or_default();
```

should we log something in case the JSON parsing fails? (this will be really challenging to debug if/when it fails, if we don't log anything and just fail silently)